commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala

```diff
@@ -8,55 +8,84 @@ import squants.motion.AngularVelocity
+                                   Dimensionless]
+}
+
+trait UnicycleHardware {
+  val forwardVelocity: PeriodicSignal[Velocity]
+  val turnVelocity: PeriodicSignal[AngularVelocity]
+}
+
 /**
   * A drivetrain that has forward-backward and turning control in the unicycle model
   */
 trait UnicycleDrive extends Drive {
   case class UnicycleSignal(forward: Dimensionless, turn: Dimensionless)
   case class UnicycleVelocity(forward: Velocity, turn: AngularVelocity)
 
-  protected def convertUnicycleToDrive(uni: UnicycleSignal): DriveSignal
-
-  protected val maxForwardVelocity: Velocity
-  protected val maxTurnVelocity: AngularVelocity
+  type DrivetrainHardware <: UnicycleHardware
+  type DrivetrainProperties <: UnicycleProperties
 
-  protected val forwardControlGains: PIDFProperUnitsConfig[Velocity, Acceleration, Length, Dimensionless]
-
-  protected val turnControlGains: PIDFConfig[AngularVelocity,
-                                             GenericValue[AngularVelocity],
-                                             GenericDerivative[AngularVelocity],
-                                             GenericIntegral[AngularVelocity],
-                                             Dimensionless]
-
-  protected val forwardVelocity: PeriodicSignal[Velocity]
-  protected val turnVelocity: PeriodicSignal[AngularVelocity]
+  /**
+    * Converts a unicycle signal value to the parent's signal type
+    * @param uni the unicycle value to convert
+    * @return the parent's equivalent signal
+    */
+  protected def convertUnicycleToDrive(uni: UnicycleSignal): DriveSignal
 
-  private def toOpenDrive(unicycle: Signal[UnicycleSignal]): Signal[DriveSignal] = {
-    unicycle.map(convertUnicycleToDrive)
+  /**
+    * Uses the parent's open loop control for the equivalent drive signal for the unicycle signal
+    * @param unicycle the unicycle signal to drive with
+    * @return a signal controlled with open-loop on the parent
+    */
+  private def parentOpenLoop(unicycle: SignalLike[UnicycleSignal]): PeriodicSignal[DriveSignal] = {
+    unicycle.map(convertUnicycleToDrive).toPeriodic
   }
 
-  private def toClosedDrive(unicycle: SignalLike[UnicycleSignal]): PeriodicSignal[DriveSignal] = {
+  /**
+    * Uses the parent's closed loop control for the drive signal for the unicycle signal
+    * @param unicycle the unicycle signal to closed-loop drive with
+    * @return a signal controlled with closed-loop on the parent
+    */
+  private def parentClosedLoop(unicycle: SignalLike[UnicycleSignal]): PeriodicSignal[DriveSignal] = {
     driveClosedLoop(unicycle.map(convertUnicycleToDrive))
   }
 
   object UnicycleControllers {
-    def forwardOpen(forwardSpeed: Signal[Dimensionless]): PeriodicSignal[DriveSignal] = {
-      toClosedDrive(forwardSpeed.map(f => UnicycleSignal(f, Percent(0))))
+    def forwardOpenLoop(forwardSpeed: Signal[Dimensionless]): PeriodicSignal[DriveSignal] = {
+      parentClosedLoop(forwardSpeed.map(f => UnicycleSignal(f, Percent(0))))
```

Comment:
That's a really good point, it's open loop at the unicycle level, but we pass the resulting speeds to closed loop at the parent (two-sided drive) level. What do you think would be the right behavior/naming here?